pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -0,0 +1,769 @@
+
+        Subscription DBus module state is represented by the SubscriptionRequest and
+        SystemPurposeData DBus structures. We first update their local mirrors from
+        the DBus module and then set all the controls in the spoke to values
+        represented in the DBus structures.
+
+        NOTE: There are a couple special cases where we need to do some special precessing,
+              such as for fields holding sensitive data. If we blindly set those based
+              on DBus structure data, we would effectively clear them as the Subscription
+              DBus module never returns previously set sensitive data in plain text.
+
+        """
+        # start by pulling in fresh data from the Subscription DBus module
+        self._subscription_request = self._get_subscription_request()
+        self._system_purpose_data = self._get_system_purpose_data()
+
+        # authentication method
+        # - SubscriptionRequest always has type set
+        # - username + password is the default
+        # For the related password and activation keys entry holding sensitive data
+        # we need to reconcile the data held in the spoke from previous entry with
+        # data set in the DBus module previously:
+        # - data in module and entry empty -> set placeholder
+        # - data in module and entry populated -> keep text in entry,
+        #   we assume it is the same as what is in module
+        # - no data in module and entry populated -> clear entry & any placeholders
+        #   (data cleared over DBus API)
+        # - no data in module and entry empty -> do nothing
+        if self.subscription_request.type == SUBSCRIPTION_REQUEST_TYPE_USERNAME_PASSWORD:
+            self.authentication_method = AuthenticationMethod.USERNAME_PASSWORD
+            self.username_entry.set_text(self.subscription_request.account_username)
+            set_in_entry = bool(self._password_entry.get_text())
+            set_in_module = self.subscription_request.account_password.type == SECRET_TYPE_HIDDEN
+            if set_in_module:
+                if not set_in_entry:
+                    self.enable_password_placeholder(True)
+            else:
+                self._password_entry.set_text("")
+                self.enable_password_placeholder(False)
+        elif self.subscription_request.type == SUBSCRIPTION_REQUEST_TYPE_ORG_KEY:
+            self.authentication_method = AuthenticationMethod.ORG_KEY
+            self._organization_entry.set_text(self.subscription_request.organization)
+            set_in_entry = bool(self._activation_key_entry.get_text())
+            set_in_module = self.subscription_request.activation_keys.type == SECRET_TYPE_HIDDEN
+            if set_in_module:
+                if not set_in_entry:
+                    self.enable_activation_key_placeholder(True)
+            else:
+                self._activation_key_entry.set_text("")
+                self.enable_activation_key_placeholder(False)
```

Comment:
This could be separated to a private method with the comment having as docstring.

Reply:
Yeah, that looks a bit better.